controller/provider/provider_payment_controller.go

```diff
@@ -0,0 +1,103 @@
+package controller
+
+import (
+	"ev-payment-service/config"
+	"ev-payment-service/dao"
+	"ev-payment-service/dto"
+	helper "ev-payment-service/helper"
+	"fmt"
+)
+
+type ProviderPaymentController interface {
+	GetProviderPaymentInfo(providerId uint, billingPeriod dto.ProviderBillingPeriod) ([]dto.ProviderPayment, error)
+	CreateProviderPayment(providerPayment *dto.ProviderPayment, token string) (string, error)
+	UpdateProviderPayment(providerPayment dto.ProviderPayment) error
+	DeleteProviderPayment(id uint, billingPeriod dto.ProviderBillingPeriod) error
+	CompleteProviderPayment(providerPayment *dto.ProviderPayment) error
+}
+
+var (
+	ProviderPaymentControllerObj ProviderPaymentController
+)
+
+type ProviderPaymentControllerImpl struct {
+	stripeKey string
+}
+
+func (p ProviderPaymentControllerImpl) GetProviderPaymentInfo(providerId uint, billingPeriod dto.ProviderBillingPeriod) ([]dto.ProviderPayment, error) {
+	providerPayment, err := dao.Db.GetProviderPaymentEntry(providerId, billingPeriod)
```

Comment:
you could have just 

return dao.Db.GetProviderPaymentEntry(providerId, billingPeriod)